Assets/Scripts/Multiplayer/MultiplayerSceneSync.cs

```diff
@@ -0,0 +1,319 @@
+            foreach (var stroke in strokes)
+            {
+                while (await MultiplayerManager.m_Instance.CheckStrokeReception(stroke, TargetPlayerId))
+                {
+                    await Task.Delay(200);
+                }
+                sentStrokes++;
+                SynchHistoryPercentage(TargetPlayerId, strokes.Count, sentStrokes);
+            }
+
+            SynchHistoryComplete(TargetPlayerId);
+        }
+
+        public async void StartSyncProgressDisplayForCommands(int TargetPlayerId, List<BaseCommand> commands)
+        {
+            StartSynchHistory(TargetPlayerId);
+
+            int sentStrokes = 0;
+            foreach (var command in commands)
+            {
+                while (await MultiplayerManager.m_Instance.CheckCommandReception(command, TargetPlayerId))
+                {
+                    await Task.Delay(200);
+                }
+                sentStrokes++;
+                SynchHistoryPercentage(TargetPlayerId, commands.Count, sentStrokes);
+            }
+
+            SynchHistoryComplete(TargetPlayerId);
+        }
+
+        private void StartSynchHistory(int id)
+        {
+            MultiplayerManager.m_Instance.StartSynchHistory(id);
+        }
+
+        private void SynchHistoryPercentage(int id, int expected, int sent)
+        {
+            MultiplayerManager.m_Instance.SynchHistoryPercentage(id, expected, sent);
+        }
+
+        private void SynchHistoryComplete(int id)
+        {
+            MultiplayerManager.m_Instance.SynchHistoryComplete(id);
+        }
+
+
+        #endregion
+
+        #region Local infoCard commands
+
+        private InfoCardAnimation infoCard;
+        private readonly object infoCardLock = new object();
+
+        public void DisplaySynchInfo()
+        {
+            lock (infoCardLock)
+            {
+                if (infoCard == null)
+                {
+                    OutputWindowScript.m_Instance.CreateInfoCardAtController(
+                        InputManager.ControllerName.Brush,
+                        "Synch Started!",
+                        fPopScalar: 1.0f
+                    );
+                    infoCard = RetrieveInfoCard();
+                }
+            }
+        }
+
+        public InfoCardAnimation RetrieveInfoCard()
+        {
+            InfoCardAnimation[] allInfoCards = FindObjectsOfType<InfoCardAnimation>();
+            foreach (var card in allInfoCards)
+            {
+                TextMeshPro textComponent = card.GetComponentInChildren<TextMeshPro>();
+                if (textComponent != null && textComponent.text.Contains("Synch"))
+                {
+                    return card;
+                }
+            }
+            return null;
+        }
+
+        public void SynchInfoPercentageUpdate()
+        {
+            lock (infoCardLock)
+            {
+                int percentage = (int)((float)SketchMemoryScript.AllStrokesCount() / numberOfCommandsExpected * 100);
```

Comment:
If people are drawing while the sync is taking place, will this still work?

Reply:
no it does not ! if someone draws the display shows the sync process goes faster 
however this is just a display  it dose not affect the sync process itself 
so yes the display or the sync progress will be affected by people drawing but it wont affect the sync process completion itself

Reply:
Gotcha. So the new strokes come in via the RPC mechanism, rather than the sync, right?

Reply:
Nope currently use SendReliableDataToPlayer. 
https://doc.photonengine.com/fusion/current/manual/data-transfer/data-streaming

However if you look at the code It depends on the SyncType
if SyncType.Strokes -> via photon SendReliableDataToPlayer not RPC  # this is the one we currently use 
if SyncType.Commands-> via RPC # this sends commands rather than strokes (one level of abstraction higher) # this i left  as photon 1 legacy or if something else might happen